pkg/controllers/management/auth/project_cluster_handler.go

```diff
@@ -113,6 +117,20 @@ func (l *clusterLifecycle) sync(key string, orig *v3.Cluster) error {
 		return err
 	}
 
+	obj, err = l.mgr.addRTAnnotation(obj, "cluster")
+	if err != nil {
+		return err
+	}
+
+	// update if it has changed
+	if obj != nil && !reflect.DeepEqual(orig, obj) {
+		logrus.Infof("[%v] Updating cluster %v", clusterCreateController, orig.Name)
+		_, err = l.mgr.mgmt.Management.Clusters("").ObjectClient().Update(orig.Name, obj)
```

Comment:
minor nit: you should not need to use the objectClient here. should be able to call Clusters("").Update(), right?